File: src/main/java/com/vaadin/demo/component/basiclayouts/BasicLayoutsHorizontalLayoutIndividualAlignment.java
package com.vaadin.demo.component.basiclayouts;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.orderedlayout.FlexComponent;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.router.Route;

@Route("basic-layouts/horizontal-layout-individual-alignment")
public class BasicLayoutsHorizontalLayoutIndividualAlignment extends Div {

    public BasicLayoutsHorizontalLayoutIndividualAlignment() {
        // tag::layout[]
        HorizontalLayout layout = new HorizontalLayout();
        layout.setPadding(true);
        layout.setAlignItems(FlexComponent.Alignment.STRETCH);

        Div item1 = new Div("Item 1");
        item1.setClassName("layout-item");
        layout.add(item1);
        layout.setAlignSelf(FlexComponent.Alignment.START, item1);

        Div item2 = new Div("Item 2");
        item2.setClassName("layout-item");
        layout.add(item2);

        Div item3 = new Div("Item 3");
        item3.setClassName("layout-item");
        layout.add(item3);
        layout.setAlignSelf(FlexComponent.Alignment.END, item3);
        // end::layout[]

        this.setClassName("basic-layouts-example");
        layout.setClassName("height-4xl");

        this.add(layout);
    }

    public static class Exporter extends // hidden-source-line
            DemoExporter<BasicLayoutsHorizontalLayoutIndividualAlignment> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/basiclayouts/basic-layouts-horizontal-layout-individual-alignment.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/horizontal-layout';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('basic-layouts-horizontal-layout-individual-alignment')
export class Example extends LitElement {
  connectedCallback() {
    super.connectedCallback();
    this.classList.add('basic-layouts-example');
  }

  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-horizontal-layout
        theme="spacing padding"
        class="height-4xl"
        style="align-items: stretch"
      >
        <div class="example-item" style="align-self: start">Item 1</div>
        <div class="example-item">Item 2</div>
        <div class="example-item" style="align-self: end">Item 3</div>
      </vaadin-horizontal-layout>
      <!-- end::snippet[] -->
    `;
  }
}
